datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1142,6 +1079,20 @@ macro_rules! binary_array_op_dyn_scalar {
     }}
 }
 
+/// Compares the array with the scalar value for equality, sometimes
+/// used in other kernels
+pub(crate) fn array_eq_scalar(lhs: &dyn Array, rhs: &ScalarValue) -> Result<ArrayRef> {
```

Comment:
I made a function to call `binary_array_op_dyn_scalar!` from `null_if` rather than having it call the macro directly to clean that code up.